internal/v5/api.go

```diff
@@ -601,6 +602,15 @@ func (h *ReqHandler) metaCharmConfig(entity *mongodoc.Entity, id *router.Resolve
 	return entity.CharmConfig, nil
 }
 
+// GET id/meta/terms
+// https://github.com/juju/charmstore/blob/v4/docs/API.md#get-idmetaterms
+func (h *ReqHandler) metaTerms(entity *mongodoc.Entity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	if entity.CharmMeta == nil || len(entity.CharmMeta.Terms) == 0 {
```

Comment:
I think this should probably return nil, nil when the entity is not a charm (which will result in an ErrMetadataNotFound error, same as when any charm-specific endpoint is accessed in a bundle). Or is the intention that terms might apply to bundles in the future too?